src/libraries/System.Net.Sockets/tests/FunctionalTests/LocalEndPointTest.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class LocalEndPointTest
+    {
+        [Fact]
+        public void UdpSocket_BoundToWildcardAddress_LocalEPDoesNotChangeOnSendTo()
+        {
+            using (Socket server = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            using (Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                server.Bind(new IPEndPoint(IPAddress.Any, 0));
+
+                Assert.Null(client.LocalEndPoint);
+
+                client.Bind(new IPEndPoint(IPAddress.Any, 0));
+
+                IPEndPoint localEPAfterBind = (IPEndPoint) client.LocalEndPoint;
+                Assert.Equal(IPAddress.Any, localEPAfterBind.Address);
+                int portAfterBind = localEPAfterBind.Port;
+
+                var sendToEP = new IPEndPoint(IPAddress.Loopback, ((IPEndPoint)server.LocalEndPoint).Port);
```

Comment:
We have the `SocketTestExtensions.BindToAnonymousPort` helper to make this kind of code simpler.

Reply:
Fixed